### ISSUE TITLE
TextLayer SDF rendering path in shader is never used

### ISSUE THREAD

**Issue by cneumann** [visgl/deck.gl#4910]
#### Description
SDF rendering for the `TextLayer` is activated by passing `fontSettings: {sdf: true}` as input props. This gets correctly propagated to the `FontAtlasManager` at [text-layer.js#L145](https://github.com/visgl/deck.gl/blob/5e480de0593ffc5b8239bbbcb7c534d76d42e61f/modules/layers/src/text-layer/text-layer.js#L145). However, at [text-layer.js#L262](https://github.com/visgl/deck.gl/blob/5e480de0593ffc5b8239bbbcb7c534d76d42e61f/modules/layers/src/text-layer/text-layer.js#L262) the `sdf` prop is extracted from `this.props` via destructuring instead of `this.props.fontSettings`. This (always `undefined`) `sdf` prop is then propagated to the `MutliIconLayer` at [text-layer.js#L277](https://github.com/visgl/deck.gl/blob/5e480de0593ffc5b8239bbbcb7c534d76d42e61f/modules/layers/src/text-layer/text-layer.js#L277). Therefore at [multi-icon-layer.js#L92](https://github.com/visgl/deck.gl/blob/5e480de0593ffc5b8239bbbcb7c534d76d42e61f/modules/layers/src/text-layer/multi-icon-layer/multi-icon-layer.js#L92) the uniform `sdf` is always false and the corresponding shader path is never used.

Additionally, passing an undocumented `sdf: true` prop to `TextLayer` activates the SDF render path, but the results are extremely ugly (see screenshot). The reason for this is that the `varying float vGamma` used to determine the smoothing zone width is zero in the fragment shader - this in turn is caused by `sizeScale` being a very large value at [multi-icon-layer.js#L45](https://github.com/visgl/deck.gl/blob/5e480de0593ffc5b8239bbbcb7c534d76d42e61f/modules/layers/src/text-layer/multi-icon-layer/multi-icon-layer.js#L45). 

#### Repro Steps
Below are my settings for `TextLayer` in 3 scenarios. The increased `fontSize`, `buffer`, and `radius` settings are only used to make the different edge smoothing behaviors easier to see.

##### Case 1
This is what should enable SDF mode for the `TextLayer`, but only affects the font atlas, not the shader path. Note that the font edge is overly blurry.
```javascript
// SDF font texture, but non-sdf shader path
new TextLayer({
    id: "text",
    data: textLayerData,
    characterSet: ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"],
    fontFamily: "sans-serif",
    fontSettings: {
        sdf: true,
        fontSize: 128,
        buffer: 10,
        radius: 8
    },
    fontWeight: "bold",
    getAlignmentBaseline: "bottom",
    getColor: color,
    getPosition: getTextPositionFn,
    getSize: 100,
    getText: getTextFn,
    getTextAnchor: "start",
});
```
![text-broken-sdf-mode](https://user-images.githubusercontent.com/117979/91882335-cc754e00-ec50-11ea-87a0-4c2b85fba912.png)

##### Case 2
Force the shaders to use the SDF code path by adding a `sdf` prop. The font has hard, non smoothed edges, not what I would expect from SDF font rendering.
```javascript
// SDF font texture, force sdf shader path
new TextLayer({
    id: "text",
    data: textLayerData,
    characterSet: ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"],
    fontFamily: "sans-serif",
    fontSettings: {
        sdf: true,
        fontSize: 128,
        buffer: 10,
        radius: 8
    },
    fontWeight: "bold",
    sdf: true,                           // NOTE: this forces shaders to actually use the SDF code path
    getAlignmentBaseline: "bottom",
    getColor: color,
    getPosition: getTextPositionFn,
    getSize: 100,
    getText: getTextFn,
    getTextAnchor: "start",
});
```
![text-force-sdf-mode](https://user-images.githubusercontent.com/117979/91882623-3f7ec480-ec51-11ea-8a96-4bd3ad00cb23.png)

##### Case 3
For comparison: turn off all SDF options. This results in a thin alpha blended edge around the glyphs.

```javascript
// regular font atlas texture and shader path
new TextLayer({
    id: "text",
    data: textLayerData,
    characterSet: ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"],
    fontFamily: "sans-serif",
    fontSettings: {
        sdf: false,              // NOTE: disable SDF mode for font atlas
        fontSize: 128,
        buffer: 10,
        radius: 8
    },
    fontWeight: "bold",
    getAlignmentBaseline: "bottom",
    getColor: color,
    getPosition: getTextPositionFn,
    getSize: 100,
    getText: getTextFn,
    getTextAnchor: "start",
});
```
![text-non-sdf-mode](https://user-images.githubusercontent.com/117979/91882973-c0d65700-ec51-11ea-9bc7-68425adb4c1d.png)

#### Environment (please complete the following information):
 - Framework Version: deck.gl 8.2.5
 - Browser Version: Chrome 85.0.4183.83
 - OS: Windows 10 x64

**Comment by Pessimistress** [comment]
Thank you for investigating this. The `sdf` uniform propagation is definitely a bug.

Regarding the visual quality: Generally speaking `fontSettings.fontSize` should be close to your output pixel size. SDF works better when upscaling, but not so much when downsampling. See [this test](https://codepen.io/Pessimistress/pen/gOrxVKw?editors=0010)

**Comment by cneumann** [comment]
Thanks for that test and I agree that SDF is generally most impressive when upscaling.
My motivation for looking into this was that I wanted to modify the TextLayer to support an outline color to improve the readability of text on top of other map elements (or busy base maps, like aerial imagery). I got something working using a custom replacement for the `MultiIconLayer` with a different fragment shader and omitting the `vGamma` calculation, instead I just use the `uniform float gamma` value - that is probably not a general solution, but I got lost between all the `size`/`sizeUnit`/`sizeScale` scaling factors and for the time being it is sufficient for my use case.
![custom-outline-text](https://user-images.githubusercontent.com/117979/91894913-61ce0d80-ec64-11ea-9f90-a0fb389c879a.png)
